### ISSUE TITLE
useMetadata kinda does not work

### ISSUE THREAD

**Issue by kremalicious** [oceanprotocol/react#61]
This will always return `undefined` or `null` for the `ddo`, so some `await` somewhere does not finish:

```js
const did = 'did:op:0x3432432432432'
const { ddo } = useMetadata(did)
```

This is for the case where `ocean` is not yet initialized. Decide if that hook is actually useful for this case, and either remove or figure out how to correctly wait for end of hook execution

**Comment by mihaisc** [comment]
The issue is probably from the `useEffect` that doesn't wait for ocean, will test proper. 
From a design point of  view i see we can have two options
1.  `useMetadata(did)`: we get the ddo and the use it for all the actions.
2. `useMetadata(ddo)` : we use the ddo provided for all the actions.
I see it like this because before with the did , we were doing so many requests for the same ddo, and those are redundant. At one point in the old market we had 3 calls to get a ddo for one metadata.
So basically if you just have a did and want to know more you use option 1, and if you have the ddo (in th AssetTeaser for example) you use option 2 because you don't need to request it again.
